### PR TITLE
rt.util.typeinfo: Deprecate typeinfo for built-in complex and imaginary types

### PR DESCRIPTION
**Pull request by ibuclaw**
Implements phase 3 of https://issues.dlang.org/show_bug.cgi?id=14488

To get around the deprecation errors when building druntime, we can use the new aliases for the "magic" complex types to build to the built-in TypeInfo classes.  These have been marked as deprecated with the provisional removal date of after version 2.105.

Druntime can now be built with `-transition=complex`.

### CODE REVIEW

File: src/rt/util/typeinfo.d [hunk 1]
```diff
@@ -36,14 +37,15 @@ if (is(T == float) || is(T == double) || is(T == real))
...
 template Floating(T)
-if (is(T == cfloat) || is(T == cdouble) || is(T == creal))
+if (is(T == c_complex_float) || is(T == c_complex_double) || is(T == c_complex_real))
 {
   pure nothrow @safe:
 
     bool equals(T f1, T f2)
     {
-        return f1 == f2;
+        return f1.re == f2.re && f1.im == f2.im;
```
Comment (RazvanN7) [inline src/rt/util/typeinfo.d:48]: Is this change really necessary? The compiler is smart enough to generate an appropriate `opEquals` for `struct Complex(T) {T re; T im;}`

File: src/rt/util/typeinfo.d [hunk 2]
```diff
@@ -105,6 +109,52 @@ if (is(T ==  float) || is(T ==  double) || is(T ==  real) ||
     public alias hashOf = core.internal.hash.hashOf;
 }
 
+template Array(T)
```
[the remark this answers is not in the document]
Comment (ibuclaw) [reply src/rt/util/typeinfo.d:112]: Makes it easy to remove later, and doesn't hinder the ability to use git-blame with useless whitespace changes.

File: src/rt/util/typeinfo.d (hunk 1 above)
Comment (ibuclaw) [reply src/rt/util/typeinfo.d:48]: This is an enum type, not a struct, so there is no `opEquals`, and there is no harm in inlining it explicitly.

File: src/rt/util/typeinfo.d (hunk 2 above)
[the remark this answers is not in the document]
Comment (RazvanN7) [reply src/rt/util/typeinfo.d:112]: Fair enough.

**Review by RazvanN7 — APPROVED** [review]

File: src/rt/util/typeinfo.d (hunk 1 above)
Comment (ibuclaw) [reply src/rt/util/typeinfo.d:48]: Actually, I'll clarify that this is only a struct type when `version=StdDdoc`.  Hence why I had to add fields to `ddoc_complex` in order to satisfy the documentation CI pipeline. :-)
